src/WindowsAzureMessaging/WindowsAzureMessaging/Models/MSInstallationTemplate.h

```diff
@@ -0,0 +1,18 @@
+//----------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//----------------------------------------------------------------
+
+#import <Foundation/Foundation.h>
+
+@interface MSInstallationTemplate : NSObject
+
+@property(nonatomic, copy) NSString *body;
+@property(nonatomic, copy) NSSet<NSString *> *tags;
+@property(nonatomic, copy) NSDictionary<NSString *, NSString *> *headers;
+
+- (void)addTag:(NSString *)tags;
```

Comment:
Why are we exposing only add tags and not remove tags? And why are we exposing set header without exposing remove header?

Reply:
Based on [template support spec](https://msmobilecenter.visualstudio.com/NotificationHub/_git/cb8c436e-73ac-48df-ab62-0252963004f5?path=%2FSDK%2F01-iOS-SDK-v2.md&_a=preview&anchor=template-support)
a client can create template with headers, tags and a body, then get the template and remove it. Should we add update template support?

Reply:
Yes, I should have been more thorough there, we need the following:

```objc
- (void)addTag:(NSString *)tag;
- (void)addTags:(NSArray<NSString *> *):tags;
- (void)removeTag:(NSString *)tag;
- (void)removeTags:(NSArray<NSString *> *):tags;
- (void)clearTags;
- (NSArray<NSString *> *)getTags;

- (void)addHeader:(NSString *)value forKey:(NSString *)key;
- (void)removeHeader(NSString *):key;
- (void)getHeader(NSString *):key;
- (void)clearHeaders;
- (NSDictionary<NSString *, NSString *> *)getHeaders;
```

Reply:
I'll fix it now for us to be the right API since I have it in my mind now.